src/lsm/lsm_cursor.c

```diff
@@ -902,6 +902,104 @@ err:	__clsm_leave(clsm);
 }
 
 /*
+ * __clsm_random_chunk --
+ *	Pick a chunk at random, weighted by the size of all chunks. Weighting
+ * proportional to documents avoids biasing towards small chunks.
+ */
+static int
+__clsm_random_chunk(WT_SESSION_IMPL *session,
+    WT_CURSOR_LSM *clsm, WT_LSM_CHUNK **random_chunk)
+{
+	WT_DECL_RET;
+	WT_LSM_TREE *lsm_tree;
+	uint64_t checked_docs, i, rand_doc, total_docs;
+
+	lsm_tree = clsm->lsm_tree;
+	rand_doc = __wt_random(&session->rnd);
```

Comment:
Since __wt_random returns a uint32_t, doesn't that mean that in a collection with huge number of docs, we skew towards finding a random doc within first set of chunks that combine to have UINT32_MAX of docs. In other words, if docs > UINT32_MAX, won't we always pick one from the beginning set of chunk that meet total docs >= UINT32_MAX criterion.